### PR TITLE
SAK-44397: Bugs fixes for SAK-42856: Provide a way to reply private

### PR DESCRIPTION
**Pull request by victorGomollon**
Bugs fixes for SAK-42856: Provide a way to reply private

### CODE REVIEW

**Comment by SedueRey** [comment]
LGTM

File: msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/PrivateMessageManagerImpl.java [hunk 1]
```diff
@@ -1184,7 +1184,7 @@ private String getSystemAndReplyEmail(String defaultEmail, User currentUser, Mes
 		    else
 			    systemEmail = ServerConfigurationService.getString(FROM_ADDRESS, defaultEmail);
 	    }
-		if (("all".equals(ServerConfigurationService.getString(REAL_REPLY, "none")) || contextId.contains(ServerConfigurationService.getString(REAL_REPLY, "none"))) && systemEmail.equalsIgnoreCase(ServerConfigurationService.getString(FROM_ADDRESS, defaultEmail))) {
+		if (ServerConfigurationService.getStrings(REAL_REPLY)!=null && ("all".equals(ServerConfigurationService.getString(REAL_REPLY, "none")) || Arrays.asList(ServerConfigurationService.getStrings(REAL_REPLY)).contains(contextId)) && systemEmail.equalsIgnoreCase(ServerConfigurationService.getString(FROM_ADDRESS, defaultEmail))) {
```
Comment (ern) [inline msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/PrivateMessageManagerImpl.java:1187]: use the ServerConfigurationService api vs the cover

File: msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/PrivateMessageManagerImpl.java [hunk 2]
```diff
@@ -1452,7 +1452,7 @@ private String buildMessageBody(PrivateMessage message) {
 	      getResourceBundleString(EMAIL_FOOTER1) + " " + ServerConfigurationService.getString("ui.service","Sakai") +
 	  " " + getResourceBundleString(EMAIL_FOOTER2) + " \"" +
 	  siteTitle + "\" " + getResourceBundleString(EMAIL_FOOTER3) + "\n";
-	  if (("all".equals(ServerConfigurationService.getString(REAL_REPLY, "none")) || contextId.contains(ServerConfigurationService.getString(REAL_REPLY, "none")))){
+	  if (ServerConfigurationService.getStrings(REAL_REPLY)!=null && ("all".equals(ServerConfigurationService.getString(REAL_REPLY, "none")) || Arrays.asList(ServerConfigurationService.getStrings(REAL_REPLY)).contains(contextId))){
```
Comment (ern) [inline msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/PrivateMessageManagerImpl.java:1455]: use the ServerConfigurationService api vs the cover

File: msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/PrivateMessageManagerImpl.java [hunk 3]
```diff
@@ -2235,7 +2235,7 @@ public void processPvtMsgReplySentAction(PrivateMessage currentMessage, PrivateM
 				  log.error(e.getMessage(), e);
 			  }
 
-			  Event event = eventTrackingService.newEvent(DiscussionForumService.EVENT_MESSAGES_RESPONSE, getEventMessage(currentMessage, DiscussionForumService.MESSAGES_TOOL_ID, rrepMsg.getAuthorId(), ((PrivateMessageRecipientImpl)currentMessage.getRecipients().get(0)).getContextId()), null, true, NotificationService.NOTI_OPTIONAL, statement);
+			  Event event = eventTrackingService.newEvent(DiscussionForumService.EVENT_MESSAGES_RESPONSE, getEventMessage(currentMessage, DiscussionForumService.MESSAGES_TOOL_ID, rrepMsg.getAuthorId(), ((PrivateMessageRecipientImpl)currentMessage.getRecipients().get(0)).getContextId()), contextId, true, NotificationService.NOTI_OPTIONAL, statement);
```
Comment (ern) [inline msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/PrivateMessageManagerImpl.java:2238]: use Types

**Review by ern — DISMISSED** [review]

File: msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/PrivateMessageManagerImpl.java [hunk 4]
```diff
@@ -1449,21 +1449,21 @@ private String buildMessageBody(PrivateMessage message) {
 	  }
 
 	  String footer = "<p>----------------------<br>" +
-	      getResourceBundleString(EMAIL_FOOTER1) + " " + ServerConfigurationService.getString("ui.service","Sakai") +
+	      getResourceBundleString(EMAIL_FOOTER1) + " " + serverConfigurationService.getString("ui.service","Sakai") +
 	  " " + getResourceBundleString(EMAIL_FOOTER2) + " \"" +
 	  siteTitle + "\" " + getResourceBundleString(EMAIL_FOOTER3) + "\n";
-	  if (("all".equals(ServerConfigurationService.getString(REAL_REPLY, "none")) || contextId.contains(ServerConfigurationService.getString(REAL_REPLY, "none")))){
+	  if (serverConfigurationService.getStrings(REAL_REPLY)!=null && ("all".equals(serverConfigurationService.getString(REAL_REPLY, "none")) || Arrays.asList(serverConfigurationService.getStrings(REAL_REPLY)).contains(contextId))){
```
Comment (ern) [inline msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/PrivateMessageManagerImpl.java:1455]: use a local variable for serverConfigurationService.getStrings(REAL_REPLY)

**Review by juanjmerono — APPROVED** [review]

**Pull request merged**